projects/storefrontlib/src/cms-components/navigation/page-header/page-title.component.ts

```diff
@@ -0,0 +1,34 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  CmsPageTitleComponent,
+  isNotNullable,
+  PageMetaService,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+
+@Component({
+  selector: 'cx-page-title',
+  templateUrl: './page-title.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PageTitleComponent implements OnInit {
+  title$: Observable<string>;
+
+  constructor(
+    public component: CmsComponentData<CmsPageTitleComponent>,
+    protected pageMetaService: PageMetaService
+  ) {}
+
+  ngOnInit(): void {
+    this.setTitle();
+  }
+
+  private setTitle(): void {
+    this.title$ = this.pageMetaService.getMeta().pipe(
+      filter(isNotNullable),
+      map((meta) => meta.heading || meta.title)
```

Comment:
I think it's possible that both are not present. Maybe add a default? Or is it not needed?

Reply:
@Pucek9 can you pls respond?

Reply:
@hackergil I didn't go deep and touch existing logic, just moved it from one class to other. If you have some proposal to improve, go ahead :)

Reply:
```suggestion
      map((meta) => (meta.heading || meta.title) ?? '')
```